### PR TITLE
Gradle upgrade + moving to MavenCentral

### PR DESCRIPTION
**Pull request by kober32**
- Upgraded Gradle scripts to Kotlin Script
- Upgraded Gradle tooling
- Scripts for MavenCentral releasing
- Added build GitHub action

### CODE REVIEW

File: Source/src_android/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        google()
+    }
+    dependencies {
+        classpath("com.android.tools.build:gradle:7.2.1")
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${Constants.kotlinVersion}")
+        // releasing
+        classpath("com.github.dcendents:android-maven-gradle-plugin:2.1")
```
Comment (TomasKypta) [inline Source/src_android/build.gradle.kts:27]: Does this really work? This plugin has been long time abandoned.
There was a ton of issues regarding compatibility with recent Gradle and AGP versions.

**Review by TomasKypta — DISMISSED** [review]
Looks OK.
Only the use of abandoned publishing plugin surprises me.

File: Source/src_android/android-release-aar.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018 Wultra s.r.o.
+ * Copyright 2021 Wultra s.r.o.
```
[the remark this answers is not in the document]
Comment (kober32) [reply Source/src_android/android-release-aar.gradle:2]: 👍🏻

File: Source/src_android/build.gradle.kts (hunk 1 above)
Comment (kober32) [reply Source/src_android/build.gradle.kts:27]: Yop, I copied it from an old project (but it still works though). Moved to the nexus publish plugin.

**Pull request merged**